vowpalwabbit/core/src/reductions/cb/details/large_action/one_pass_svd_impl.cc

```diff
@@ -77,8 +96,15 @@ class AO_triplet_constructor
   uint64_t _column_index;
   uint64_t _seed;
   float& _final_dot_product;
+  // index mapped used to select sparsity or not from value map
+  constexpr static std::array<size_t, 2> index_map = {0, 2};
+  constexpr static std::array<float, 4> value_map = {0.f, 0.f, 1.f, -1.f};
 };
 
+// definition at namespace scope
+constexpr std::array<size_t, 2> AO_triplet_constructor::index_map;
+constexpr std::array<float, 4> AO_triplet_constructor::value_map;
```

Comment:
What are these lines for?

Reply:
you need to declare the constexpr static members at namespace scope ([resolved automatically in c++17](https://en.cppreference.com/w/cpp/language/static#Constant_static_members))

Reply:
and can not declare them inline because (and I quote) "inline variables are only available with c++17"